src/lib.rs

```diff
@@ -259,6 +261,29 @@ mod contract {
         result.map(|_f| Vec::new()).sdk_process();
     }
 
+    #[no_mangle]
+    pub extern "C" fn register_relayer() {
+        let relayer_address = sdk::read_input();
+        let mut engine = Engine::new(predecessor_address());
+        engine.register_relayer(
+            sdk::predecessor_account_id().as_slice(),
+            Address(relayer_address.as_slice().try_into().unwrap()),
+        );
+    }
+
+    /// Allow receiving NEP141 tokens to the EVM contract
+    #[no_mangle]
+    pub extern "C" fn ft_on_transfer() {
+        let predecessor_account_id = String::from_utf8(sdk::predecessor_account_id()).unwrap();
+        let current_account_id = String::from_utf8(sdk::current_account_id()).unwrap();
+
+        if current_account_id == predecessor_account_id {
+            // TODO(#59) ETH transfer
+        } else {
+            // TODO(#51) ERC20 transfer
+        }
+    }
```

Comment:
Obviously, clippy is going to scream about this. Can you please instead add in `todo!()` in both?